tests/InputFiles/GeneralizedHarmonic/GaugeWave1D.yaml

```diff
@@ -39,13 +44,12 @@ DomainCreator:
          InitialTime: 0.0
          Velocity: [0.5]
     BoundaryConditions:
-      LowerBoundary:  DirichletAnalytic
-      UpperBoundary:  DirichletAnalytic
-
-AnalyticSolution:
-  GaugeWave:
-    Amplitude: 0.5
-    Wavelength: 6.283185307179586
+      LowerBoundary:
+        DirichletAnalytic:
+          AnalyticPrescription: *InitialData
```

Comment:
Not a change request, but this feels like it can be broken pretty easily. Some poor soul (me) copies the ID instead of using the YAML `& -> *` syntax, then changes the ID but forgets to update the BCs and bad things happen.

Do you think there'd be a way to have DirichletAnalytic retrieve the ID somehow on its own so users don't have to worry about duplicating the ID parameters (YAML or otherwise)? If not, or it isn't trivial, don't worry about it, but I can imagine some issues arising because of this.

Reply:
This is intentional. There is generally no reason why you should apply the same BC as the initial condition. The current develop forces this on users and it's been a bit annoying, e.g. see the Minkowski BC Geoffrey wrote. Being able to specify them separately is both reasonable and necessary. I agree it adds an avenue for bugs, but I don't see a way to prevent that while adding the flexibility. The best I've got is have the example input files use the `&` and `*` syntax :)

Reply:
Ok that makes sense. This is alright then.